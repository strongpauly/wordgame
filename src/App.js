import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import Game from './Game';
import propTypes from 'prop-types';

export default class App extends Component {

  static propTypes = {
    width: propTypes.number,
    height: propTypes.number,
  }

  constructor(props) {
    super(props);
    this.restart = this.restart.bind(this);
    this.state = {
      width: props.width || 6,
      height: props.height || 6,
      gameId: 0
    };
  }

  restart() {
    this.setState({gameId: this.state.gameId + 1});
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo spinning" alt="logo" />
          <h2>Word Game!</h2>
        </div>
        <Game key={this.state.gameId}
            gameId={this.state.gameId}
            width={this.state.width}
            height={this.state.height}
            onRestart={this.restart}></Game>
      </div>
    );
  }
}
