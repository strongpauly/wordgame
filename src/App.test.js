import React from 'react';
import App from './App';
import {shallow, mount} from 'enzyme';

/* eslint-env jest */

describe('<App>', () => {
  it('renders without crashing', () => {
    mount(<div><App /></div>);
  });

  it('creates a new game id when restarted.', () => {
    const app = shallow(<App/>);
    expect(app).toMatchSnapshot();
    expect(app.find('Game[gameId=0]')).toHaveLength(1);
    app.instance().restart();
    expect(app.find('Game[gameId=0]')).toHaveLength(0);
    expect(app.find('Game[gameId=1]')).toHaveLength(1);
  });

  // it('clears completed when restarted.', () => {
  //   const app = mount(<App width={1} height={2}/>);
  //   //Not completed
  //   expect(app.find('.completed')).toHaveLength(0);
  //   //Click one cell - this should either explode or win the game as there are only two cells.
  //   app.find('Cell td').at(0).simulate('click');
  //   //Game should be completed
  //   expect(app.find('.completed')).toHaveLength(1);
  //   //Restart the game.
  //   app.instance().restart();
  //   //Game should be reset.
  //   expect(app.find('.completed')).toHaveLength(0);
  // });
});
