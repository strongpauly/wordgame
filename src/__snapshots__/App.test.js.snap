// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> creates a new game id when restarted. 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo spinning"
      src="logo.svg"
    />
    <h2>
      Word Game!
    </h2>
  </div>
  <Game
    gameId={0}
    height={6}
    onRestart={[Function]}
    width={6}
  />
</div>
`;
